src/main/java/org/owasp/benchmark/score/report/ScatterPlot.java

```diff
@@ -0,0 +1,246 @@
+        NumberAxis rangeAxis = (NumberAxis) xyplot.getRangeAxis();
+        NumberAxis domainAxis = (NumberAxis) xyplot.getDomainAxis();
+
+        rangeAxis.setRange(-9.99, 109.99);
+        rangeAxis.setNumberFormatOverride(pctFormat);
+        rangeAxis.setTickLabelPaint(Color.decode("#666666"));
+        rangeAxis.setMinorTickCount(5);
+        rangeAxis.setTickUnit(new NumberTickUnit(10));
+        rangeAxis.setAxisLineVisible(true);
+        rangeAxis.setMinorTickMarksVisible(true);
+        rangeAxis.setTickMarksVisible(true);
+        rangeAxis.setLowerMargin(10);
+        rangeAxis.setUpperMargin(10);
+        
+        domainAxis.setRange(-5, 175);
+        domainAxis.setNumberFormatOverride(pctFormat);
+        domainAxis.setTickLabelPaint(Color.decode("#666666"));
+        domainAxis.setMinorTickCount(5);
+        domainAxis.setTickUnit(new NumberTickUnit(10));
+        domainAxis.setAxisLineVisible(true);
+        domainAxis.setTickMarksVisible(true);
+        domainAxis.setMinorTickMarksVisible(true);
+        domainAxis.setLowerMargin(10);
+        domainAxis.setUpperMargin(10);
+        
+        xyplot.setRangeGridlineStroke(new BasicStroke());
+        xyplot.setRangeGridlinePaint(Color.lightGray);
+        xyplot.setRangeMinorGridlinePaint(Color.decode("#DDDDDD"));
+        xyplot.setRangeMinorGridlinesVisible(true);
+        xyplot.setOutlineVisible(true);
+        xyplot.setDomainGridlineStroke(new BasicStroke());
+        xyplot.setDomainGridlinePaint(Color.lightGray);
+        xyplot.setDomainMinorGridlinePaint(Color.decode("#DDDDDD"));
+        xyplot.setDomainMinorGridlinesVisible(true);
+        xyplot.getRenderer().setSeriesPaint(0, Color.decode("#4572a7"));
+
+        chart.setTextAntiAlias(true);
+        chart.setAntiAlias(true);
+        chart.removeLegend();
+        chart.setPadding(new RectangleInsets(20, 20, 20, 20));
+        
+        Point2D legendLocation = new Point2D.Double( 101, -10 );
+        makeRect(xyplot, legendLocation, 120, 74, Color.WHITE );
+       
+        Point2D triangleLocation = new Point2D.Double( 101, -10 );
+        Color grey = new Color(0.1f,0.1f,0.1f,0.1f);
+        makeTriangle(xyplot, triangleLocation, grey );
+        
+        makeGuessingLine( xyplot );
+    }
+    
+    
+    public static XYPointerAnnotation makePointer(int x, int y, String msg, TextAnchor anchor, int angle ) {
+        XYPointerAnnotation pointer = new XYPointerAnnotation(msg, x, y, Math.toRadians(angle));
+        pointer.setBackgroundPaint(Color.white);
+        pointer.setTextAnchor(anchor);
+        pointer.setArrowWidth(4);
+        pointer.setArrowLength(8);
+        pointer.setArrowPaint(Color.red);
+        pointer.setLabelOffset(2);
+        pointer.setPaint(Color.red);
+        pointer.setFont(theme.getRegularFont());
+        return pointer;
+    }
+    
+    public void writeChartToFile(File f, int height) throws IOException {
+        FileOutputStream stream = new FileOutputStream(f);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *stream*. The resource is referenced in statements at the following line: 146. The resource closure statement is at line: 147. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by ChartUtils.writeChartAsPNG. Either a) close *stream* in a try-finally block or b) close the resource by declaring *stream* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).